This fails. So we can't use `dropna` in `IndexOpsMixin` even if this is inherited by both index and series which have `dropna` method? @arw2019 
```
Performing static analysis using mypy
pandas/core/base.py:1035: error: "IndexOpsMixin" has no attribute "dropna"  [attr-defined]
```

--- a/doc/source/whatsnew/v1.2.0.rst
+++ b/doc/source/whatsnew/v1.2.0.rst
@@ -562,7 +562,7 @@ Other
 - Bug in :meth:`Index.union` behaving differently depending on whether operand is a :class:`Index` or other list-like (:issue:`36384`)
 - Passing an array with 2 or more dimensions to the :class:`Series` constructor now raises the more specific ``ValueError``, from a bare ``Exception`` previously (:issue:`35744`)
 - Bug in ``accessor.DirNamesMixin``, where ``dir(obj)`` wouldn't show attributes defined on the instance (:issue:`37173`).
-- Bug in :meth:`Series.nunique` with ``dropna = True`` was returning incorrect results when both ``NA`` and ``None`` missing values were present (:issue:`37566`)
+- Bug in :meth:`Series.nunique` with ``dropna=True`` was returning incorrect results when both ``NA`` and ``None`` missing values were present (:issue:`37566`)
 
 .. ---------------------------------------------------------------------------
 

--- a/pandas/core/base.py
+++ b/pandas/core/base.py
@@ -33,7 +33,7 @@ from pandas.core.dtypes.common import (
     is_scalar,
 )
 from pandas.core.dtypes.generic import ABCDataFrame, ABCIndexClass, ABCSeries
-from pandas.core.dtypes.missing import isna
+from pandas.core.dtypes.missing import isna, remove_na_arraylike
 
 from pandas.core import algorithms
 from pandas.core.accessor import DirNamesMixin
@@ -1032,7 +1032,7 @@ class IndexOpsMixin(OpsMixin):
         >>> s.nunique()
         4
         """
-        obj = self.dropna() if dropna else self
+        obj = remove_na_arraylike(self) if dropna else self
         return len(obj.unique())
 
     @property

--- a/pandas/tests/base/test_unique.py
+++ b/pandas/tests/base/test_unique.py
@@ -127,5 +127,5 @@ def test_unique_bad_unicode(idx_or_series_w_bad_unicode):
 def test_nunique_dropna(dropna):
     # GH37566
     s = pd.Series(["yes", "yes", pd.NA, np.nan, None, pd.NaT])
-    res = s.nunique(dropna=True)
+    res = s.nunique(dropna)
     assert res == 1 if dropna else 5

